packages/firestore/src/remote/bloom_filter.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Md5, Integer } from '@firebase/webchannel-wrapper';
+
+import { debugAssert } from '../util/assert';
+
+export class BloomFilter {
+  private readonly bitSize: number;
+
+  constructor(
+    private readonly bitmap: Uint8Array,
+    padding: number,
+    private readonly hashCount: number
+  ) {
+    this.bitSize = this.bitmap.length * 8 - padding;
+    if (this.bitSize === 0) {
+      debugAssert(
+        this.hashCount === 0,
+        'An empty bitmap should correspond to 0 hashCount.'
+      );
+    } else {
+      debugAssert(padding >= 0, 'Padding is negative.');
+      debugAssert(this.bitSize >= 0, 'Bitmap size is negative.');
+      debugAssert(this.hashCount >= 0, 'Hash count is negative.');
+    }
+  }
+
+  getBitSize(): number {
+    return this.bitSize;
+  }
+
+  mightContain(document: string): boolean {
+    // Empty bitmap should always return false on membership check
+    if (this.bitSize === 0) {
+      return false;
+    }
+
+    // Hash the string using md5
+    const md5 = new Md5();
+    md5.update(document);
+    const encodedBytes: Uint8Array = new Uint8Array(md5.digest());
+
+    // Interpret the hashed value as two 64-bit chunks as unsigned integers, encoded using 2’s
+    // complement using little endian.
+    const dataView = new DataView(encodedBytes.buffer);
+    const firstUint32 = dataView.getUint32(0, /* littleEndian= */ true);
```

Comment:
DataView.getUint32 browser compatibility is pretty good: https://caniuse.com/?search=getUint32 . Is this okay to use here?

Reply:
Yes, I think we're good to use `DataView.getUint32()`.

We already use `setFloat64()`: https://github.com/firebase/firebase-js-sdk/blob/fde5adf638dae4714ff7f25c75e475344907e05e/packages/firestore/src/index/ordered_code_writer.ts#L45

And `setFloat64()` has the same availability as `getUint32()`.